blockchain/state_manager/src/lib.rs

```diff
@@ -122,6 +138,83 @@ where
         Ok((state_root, rect_root))
     }
 
+    pub async fn tipset_state(&self, tipset: &Tipset) -> Result<(Cid, Cid), Box<dyn StdError>> {
+        span!("tipset_state", {
+            trace!("tipset {:?}", tipset.cids());
+            // if exists in cache return
+            if let Some(cid_pair) = self.cache.read().await.get(&tipset.key()) {
+                return Ok(cid_pair.clone());
+            }
+
+            if tipset.len() == 0 {
```

Comment:
tipset.is_empty()

Reply:
hmmm I think the is_empty() is not a function on the tipset struct, checking right now

Reply:
yeah, it isn't. dw about that now